GVFS/GVFS.Common/Tracing/EventListener.cs

```diff
@@ -14,30 +14,27 @@ protected EventListener(EventLevel maxVerbosity, Keywords keywordFilter)
             this.keywordFilter = keywordFilter;
         }
 
+        public event EventHandler ListenerRecovery;

```

Comment:
Because the posting of events over the pipe is now disconnected from actual tracer message write, failure (and recovery) is now communicated using events.

Reply:
We're using an event-sink object pattern rather than events now.